sdk/include/simulator.h

```diff
@@ -3,14 +3,18 @@
 
 #pragma once
 #include <compartment.h>
+#include <errno.h>
 #include <stdint.h>
 
 #ifdef SIMULATION
 /**
  * Exit simulation, reporting the error code given as the argument.
  */
-[[cheri::interrupt_state(disabled)]] void __cheri_compartment("sched")
+[[cheri::interrupt_state(disabled)]] int __cheri_compartment("sched")
```

Comment:
We can just silence the warning here, I think.  Or we can rename this to `scheduler_exit_simulation` and have a `void` wrapper that calls it.